src/SmartHouse/SmartHouse.ts

```diff
@@ -0,0 +1,89 @@
+import Logger from "../Utilities/Logger/Logger";
+import NameValidityChecker from '../shared/NameValidityChecker/NameValidityChecker';
+import { DeviceInterface } from '../BaseDevices/Device/types';
+import * as Collections from 'typescript-collections';
+
+export class SmartHouse<T extends DeviceInterface> {
+    private name: string;
+    private devices: Collections.Dictionary<string, T>;
+
+    constructor(name: string = 'Smart House') {
+        if (NameValidityChecker(name)) {
+            this.name = name;
+            this.devices = new Collections.Dictionary<string, T>();
+        }
+    };
+
+    onAll(): void {
+        this.devices.forEach( (deviceName, device) => device.on());
+    };
+
+    offAll(): void {
+        this.devices.forEach((deviceName, device) => device.off());
+    };
+
+    deleteAllDevices(): void {
+        this.offAll();
+        this.devices.clear();        
+    };
+
+    getAllDevicesByModel( Model: Function ): Array<T> | [] {
```

Comment:
you have a very generic method signature. Also don't see any reason to specify empty array there